src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx

```diff
@@ -170,7 +170,13 @@ export default class ReviewCollapsibleChapter extends React.Component {
                       )
                     }
                     onChange={formData =>
-                      this.onChange(formData, page.arrayPath, page.index)
+                      this.onChange(
+                        typeof page.updateFormData === 'function'
```

Comment:
Think there's a way to get this logic into a function that can be shared? It's in three different places now in a similar-ish form. But they might not quite be close enough.

Reply:
Yeah, like we talked about earlier, we would ideally be able to consolidate this in the `SET_DATA` reducer, but I think there's a lot more overhead with that approach. (Getting the `updateFormData` function to only fire when data on its page changes—or just always fire all `updateFormData` functions for all pages on all data changes :grimacing:—getting the old data separate from the new data, that kind of thing.) I didn't really look too much into it, though.

I'm hoping with sufficient testing, it'll be alright. :crossed_fingers: